packages/utils/node-resolver-rs/src/specifier.rs

```diff
@@ -136,8 +136,9 @@ impl<'a> Specifier<'a> {
             }
           }
           SpecifierType::Cjs => {
-            if BUILTINS.contains(&specifier) {
-              (Specifier::Builtin(Cow::Borrowed(specifier)), None)
+            let builtin = specifier.strip_prefix("node:").unwrap_or(specifier);
+            if BUILTINS.contains(&builtin) {
+              (Specifier::Builtin(Cow::Borrowed(builtin)), None)
```

Comment:
If `specifier` starts with `"node:"` then it should always be treated as a builtin, that was why **Node** introduced it in the first place (and also parity with **ESM**):
- https://github.com/nodejs/node/issues/36098
- https://github.com/nodejs/node/pull/37246